src/librbd/io/ImageRequestWQ.cc

```diff
@@ -557,6 +588,14 @@ void *ImageRequestWQ<I>::_void_dequeue() {
     ThreadPool::PointerWQ<ImageRequest<I> >::_void_dequeue());
   assert(peek_item == item);
 
+  if (item->needs_throttle() && iops_throttle->get(1, new C_Throttle(this, item))) {
+    // io was queued into blockers list and wait for tokens.
+    this->get_pool_lock().Unlock();
+    this->signal();
+    this->get_pool_lock().Lock();
```

Comment:
Increment ```m_io_blockers``` before returning to block other IO operations -- the callback from the throttle should decrement it, push it back into the queue, and signal the thread to wake up

Reply:
okey

Reply:
wait a minute, should we block the other IO operations?

Reply:
This is a global IOPS throttle -- so if you've hit the limit for the time period, no other IO operations should be running